### PR TITLE
debounce works with share and exshare

### PR DESCRIPTION
**Pull request by Shahar-Y**
Signed-off-by: shahar-y <shya95@gmail.com>

### CODE REVIEW

File: src/components/files/AutoSuggestor.vue [hunk 1]
```diff
@@ -75,44 +77,53 @@ export default {
...
+      if (!this.selected) return;
+      this.$emit("select", { value: this.selected });
     }
   }
 };
+
+// An async-debouncer function. 
+// The 'func' is called after the requested interval
+function asyncDebouncer(func, interval) {
+  console.log(interval);
```
Comment (yonatandt) [inline src/components/files/AutoSuggestor.vue]: delete console.log

File: src/components/files/AutoSuggestor.vue [hunk 2]
```diff
@@ -75,44 +77,53 @@ export default {
...
+      const res = await usersApi.searchUserByName(this.input);
       const users = res.data.users;
       if (users) {
         this.suggestions = [{ data: users }];
       }
       return users ? users : [];
+    },
+    submitSelected() {
+      if (!this.selected) return;
+      this.$emit("select", { value: this.selected });
```
Comment (yonatandt) [inline src/components/files/AutoSuggestor.vue:106]: Clear the search-bar input after this event

File: src/components/files/AutoSuggestor.vue (hunk 1 above)
Comment (Shahar-Y) [reply src/components/files/AutoSuggestor.vue]: done.

File: src/components/files/AutoSuggestor.vue (hunk 2 above)
Comment (Shahar-Y) [reply src/components/files/AutoSuggestor.vue:106]: done.

**Pull request merged**